tests/integration/Tribe/Language_Test.php

```diff
@@ -0,0 +1,141 @@
+<?php
+
+namespace Tribe;
+
+use Closure;
+use Codeception\TestCase\WPTestCase;
+use MO;
+use PO;
+use Tribe__Main;
+use Tribe__Events__Main;
+
+class Language_Test extends WPTestCase {
+
+	/**
+	 * @dataProvider load_text_domain_data_provider
+	 * @test
+	 */
+	public function it_should_load_text_domain_based_on_conditions( Closure $setup, string $expected_result, ?string $expected_message = null ): void {
+		// Execute the setup logic and retrieve the result.
+		$result = $setup();
```

Comment:
I would also like to evaluate how many times we call `load_plugin_textdomain` in every case. Thats where my comments on prod code are aimed for.

if the first call of load_plugin_textdomain is good there shouldn't be a second.

Can you try to test that scenario ?